WcaOnRails/app/views/users/edit.html.erb

```diff
@@ -186,8 +186,10 @@
       <div class="tab-pane active" id="preferences">
         <%= simple_form_for @user do |f| %>
           <%= hidden_field_tag "section", "preferences" %>
-          <%= f.input :preferred_event_ids, as: :events_picker, allowed_events: Event.all_official, selected_events: @user.preferred_events,
-                                            disabled: !editable_fields.include?(:preferred_event_ids), hint: "Selecting your preferred events makes registration for competitions quicker." %>
+
+          <%= render "shared/associated_events_picker", form_builder: f, label: "Preferred events", disabled: ! editable_fields.include?(:preferred_events),
```

Comment:
super-nit: I see a space after this exclamation mark. I don't think we've been doing that elsewhere.